File: jlpt/native_android/app/src/main/java/com/jlpt/android/domain/repository/ContentRepository.kt
package com.jlpt.android.domain.repository

import com.jlpt.android.domain.model.Grammar
import com.jlpt.android.domain.model.JlptLevel
import com.jlpt.android.domain.model.Kanji
import com.jlpt.android.domain.model.ListeningAudio
import com.jlpt.android.domain.model.Passage
import com.jlpt.android.domain.model.Vocabulary
import com.jlpt.android.domain.util.AppResult

interface ContentRepository {
    suspend fun getVocabulary(level: JlptLevel): AppResult<List<Vocabulary>>
    suspend fun getKanji(level: JlptLevel): AppResult<List<Kanji>>
    suspend fun getGrammar(level: JlptLevel): AppResult<List<Grammar>>
    suspend fun getPassage(id: Long): AppResult<Passage>
    suspend fun getListeningAudio(id: Long): AppResult<ListeningAudio>
}

File: jlpt/native_ios/Sources/Domain/Repositories/ContentRepository.swift
import Foundation

protocol ContentRepository {
    func getVocabulary(level: JlptLevel) async -> AppResult<[Vocabulary]>
    func getKanji(level: JlptLevel) async -> AppResult<[Kanji]>
    func getGrammar(level: JlptLevel) async -> AppResult<[Grammar]>
    func getPassage(id: Int64) async -> AppResult<Passage>
    func getListeningAudio(id: Int64) async -> AppResult<ListeningAudio>
}
